AdvancedReports.php

```diff
@@ -1020,6 +1328,58 @@ function outputViewReportJS()
 <script type="text/javascript">
   $(function()
   {
+<?php
+
+
+		$userDateFormat = \DateTimeRC::get_user_format_base();
+		if ( $userDateFormat == 'DMY' || $userDateFormat == 'MDY' )
+		{
+			$userDateSubstr = $userDateFormat == 'DMY' ? [ '3,5', '0,2' ] : [ '0,2', '3,5' ];
+?>
+    var vDateParse = Date.parse
+    Date.parse = function ( vDateVal )
+    {
+      if ( /^[0-9]{2}[^0-9][0-9]{2}[^0-9][0-9]{4}([^0-9]|$)/.test( vDateVal ) )
```

Comment:
is this meant to be a date parsing regex, if so it doesn't look quite right to me.  Separators?  Years should start with a 19, 20 or 21 potentially.

Reply:
The regex is only meant to capture anything which sort of resembles a date in dmy or mdy format. It will rearrange it to ymd format for the JavaScript date parser. If it's not a valid date the parser will pick this up anyway.